### PR TITLE
Pks fix bug1784

### PR DESCRIPTION
**Pull request by pavithraks**
This is a fix to bug 1784 (https://github.com/VowpalWabbit/vowpal_wabbit/issues/1784)

The problem was that the actual scores list was indexed by action. In the python binding, in `pyvw.cc`, in the function `ex_get_action_scores`, we were appending the probabilities to a list ignoring the index that specified the action. I have added a new array that stores both the index and the score and use this new array to generate the list that is returned.

### CODE REVIEW

**Comment by jackgerrits** [comment]
It looks like your other bug fix is in this branch too, could you rebase onto master?

File: python/pylibvw.cc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #include "../vowpalwabbit/vw.h"
-
+#include<iostream>
```
Comment (jackgerrits) [inline python/pylibvw.cc]: remove

**Comment by jackgerrits** [comment]
Since the array needs to be dynamically sized, I'd suggest using a vector here.

```c++
py::list ex_get_action_scores(example_ptr ec)
{
  py::list values;
  v_array<ACTION_SCORE::action_score> scores = ec->pred.a_s;
  std::vector<float> ordered_scores(scores.size());

  for (auto action_score : scores)
  {
    ordered_scores[action_score.action] = action_score.score;
  }

  for(auto action_score: ordered_scores)
  {
    values.append(action_score);
  }

  return values;
}
```

**Comment by pavithraks** [comment]
@jackgerrits  If this is a stylistic issue, then I'm happy to make the change. I don't see the point of dynamic allocation given how it's being used in this case. We know the length we need before hand and there are no surprises here. Why then do we need the ability for dynamic allocation?

**Comment by jackgerrits** [comment]
It's not a style comment, this is a real issue. When I say dynamic, I mean it is known at runtime and not compile time. The way the array is used right now is not correct

**Comment by pavithraks** [comment]
Oh! I see what you mean. Thanks! I'll make the change :)

**Comment by jackgerrits** [comment]
If you're compiling with GCC it has an extension to enable this behavior but it's a bit funky and not supported everywhere, so we don't want it here

File: python/pylibvw.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackgerrits) [reply python/pylibvw.cc]: This include needs to be removed as it is no longer used

File: vowpalwabbit/bfgs.cc [hunk 2]
```diff
@@ -457,18 +457,19 @@ double add_regularization(vw& all, bfgs& b, float regularization, T& weights)
     }
 
   // if we're not regularizing the intercept term, then subtract it off from the result above
+  // when accessing weights[constant], always use weights.strided_index(constant)
```
[the remark this answers is not in the document]
Comment (pavithraks) [reply vowpalwabbit/bfgs.cc]: yeah, I did not realize that I branched off my old branch. Will fix it with my next commit

**Pull request closed without merging**